apps/web-client/src/lib/db/cr-sqlite/__tests__/reconciliation-order.test.ts

```diff
@@ -1321,3 +1323,204 @@ describe("Misc helpers", () => {
+				`Cannot delete finalized reconciliation order ${reconciliationOrderId}`
+			);
+		});
+	});
+	describe("deleteOrderLineFromReconciliationOrder", () => {
+		it("should delete a specific order line from reconciliation order", async () => {
+			const db = await getRandomDb();
+
+			// Create supplier order
+			await addBooksToCustomer(db, 1, ["1"]);
+
+			await createSupplierOrder(db, 1, 1, [{ isbn: "1", quantity: 1, supplier_id: 1 }]);
+			const [{ id: supplierOrderId }] = await getPlacedSupplierOrders(db);
+
+			// Create reconciliation order
+			const reconciliationOrderId = 123;
+			await createReconciliationOrder(db, reconciliationOrderId, [supplierOrderId]);
+			await addOrderLinesToReconciliationOrder(db, reconciliationOrderId, [
+				{ isbn: "1", quantity: 2 },
+				{ isbn: "2", quantity: 1 }
+			]);
+
+			// Get initial state
+			const initialLines = await getReconciliationOrderLines(db, reconciliationOrderId);
+			expect(initialLines).toHaveLength(2);
+
+			// Delete one line
+			await deleteOrderLineFromReconciliationOrder(db, reconciliationOrderId, "1");
+
+			// Verify deletion
+			const remainingLines = await getReconciliationOrderLines(db, reconciliationOrderId);
+			expect(remainingLines).toHaveLength(1);
+			expect(remainingLines[0].isbn).toBe("2");
+		});
+
+		it("should update the reconciliation order's timestamp when deleting a line", async () => {
+			const db = await getRandomDb();
+
+			// Create supplier order
+			await addBooksToCustomer(db, 1, ["1"]);
+			await createSupplierOrder(db, 1, 1, [{ isbn: "1", quantity: 1, supplier_id: 1 }]);
+			const [{ id: supplierOrderId }] = await getPlacedSupplierOrders(db);
+
+			// Create reconciliation order
+			const reconciliationOrderId = 123;
+
+			await createReconciliationOrder(db, reconciliationOrderId, [supplierOrderId]);
+			await addOrderLinesToReconciliationOrder(db, reconciliationOrderId, [{ isbn: "1", quantity: 1 }]);
+
+			// Get initial timestamp
+			const initialOrder = await getReconciliationOrder(db, reconciliationOrderId);
+			const initialTimestamp = initialOrder.updatedAt;
+
+			// Wait a bit to ensure timestamp difference
+			await new Promise((resolve) => setTimeout(resolve, 100));
+
+			// Delete the line
+			await deleteOrderLineFromReconciliationOrder(db, reconciliationOrderId, "1");
+
+			// Check updated timestamp
+			const updatedOrder = await getReconciliationOrder(db, reconciliationOrderId);
+			expect(updatedOrder.updatedAt.getTime()).toBeGreaterThan(initialTimestamp.getTime());
+		});
+
+		it("should throw error when reconciliation order doesn't exist", async () => {
+			const db = await getRandomDb();
+
+			const nonExistentId = 999;
+			await expect(deleteOrderLineFromReconciliationOrder(db, nonExistentId, "1")).rejects.toThrow(
+				`Reconciliation order ${nonExistentId} not found or already finalized`
+			);
+		});
+
+		it("should not throw when deleting non-existent ISBN from valid order", async () => {
```

Comment:
It shouldn't throw?

Reply:
should it throw?

Reply:
I think something should be returned to the caller. Does the transaction fail if a matching row isn't found?

Reply:
no